test/gtest/unit_FinInterface.cpp

```diff
@@ -0,0 +1,620 @@
+        os << ")";
+        return os;
+    }
+
+    std::string algo;
+};
+
+using BatchNormSolverInfo = SolverInfo;
+
+struct SolverConfig
+{
+    SolverConfig() : empty(true) {}
+    SolverConfig(bool empty_) : empty(empty_) {}
+
+    bool empty;
+};
+
+struct ConvSolverConfig : SolverConfig, private miopen::unit_tests::ConvTestCase
+{
+    ConvSolverConfig(miopen::conv::Direction direction_,
+                     miopen::unit_tests::TensorDescriptorParams&& x,
+                     miopen::unit_tests::TensorDescriptorParams&& w,
+                     miopenDataType_t type_y,
+                     miopen::unit_tests::ConvolutionDescriptorParams&& conv)
+        : SolverConfig(false),
+          miopen::unit_tests::ConvTestCase(std::move(x), std::move(w), type_y, std::move(conv)),
+          direction(direction_)
+    {
+    }
+
+    auto GetProblemDescription() const { return GetProblemDescription(direction); }
+
+    friend std::ostream& operator<<(std::ostream& os, const ConvSolverConfig& config)
+    {
+        os << "(";
+        if(config.empty)
+        {
+            os << "empty";
+        }
+        else
+        {
+            os << "direction:" << static_cast<int>(config.direction);
+            os << ", " << static_cast<const miopen::unit_tests::ConvTestCase&>(config);
+        }
+        os << ")";
+        return os;
+    }
+
+private:
+    miopen::conv::Direction direction;
+
+    using miopen::unit_tests::ConvTestCase::GetProblemDescription;
+    using SolverConfig::SolverConfig;
+};
+
+struct BatchNormSolverConfig : SolverConfig
+{
+    BatchNormSolverConfig(int dummy) : SolverConfig(false) { std::ignore = dummy; }
+
+    auto GetProblemDescription() const
+    {
+        return miopen::batchnorm::ProblemDescription{{}, {}, {}, {}, {}, {}, {}, {}};
+    }
+
+    friend std::ostream& operator<<(std::ostream& os, const BatchNormSolverConfig& config)
+    {
+        os << "(";
+        if(config.empty)
+            os << "empty";
+        else
+            os << "none";
+        os << ")";
+        return os;
+    }
+
+    using SolverConfig::SolverConfig;
+};
+
+template <class Info, class SolverConfig>
+struct TestCase
+{
+    friend std::ostream& operator<<(std::ostream& os, const TestCase& tc)
+    {
+        os << "(";
+        os << "name:" << tc.name;
+        os << ", info:" << tc.info;
+        os << ", config:" << tc.config;
+        os << ")";
+        return os;
+    }
+
+    std::string name;
+    Info info;
+    SolverConfig config;
+};
+
+using ConvTestCase      = TestCase<ConvSolverInfo, ConvSolverConfig>;
+using BatchNormTestCase = TestCase<BatchNormSolverInfo, BatchNormSolverConfig>;
+
+const auto& GetTestParams()
+{
+    static const auto params = TestParams{};
+    return params;
+}
+
+template <class SolverInfo>
+const auto& GetSolversInfo();
+
+template <>
+const auto& GetSolversInfo<ConvSolverInfo>()
+{
+    static const std::unordered_map<std::string, ConvSolverInfo> solver_info = {
```

Comment:
Wow, what a list! Is this generated, or maintained by hand? There are likely to be many additions coming with MX data types. Seems like it wouldn't be too hard to import it from a file that would be easier for external tools to interact with (e.g. json, csv).

In either case, it would be nice to have placeholders for the missing ordinals. That way we can know at a glance if an algo was deleted or otherwise expected to be missing. Otherwise, it looks like something may have been missed.

Reply:
This list is filled manually. It contains all the solvers for the specified primitives currently available in the library. This is needed to check that the interface provides correct data. However, we don't need such a large list; a dozen solvers would be sufficient. There is also no need to update it when new solvers are added.